templates/package/content/arch.tmpl

```diff
@@ -0,0 +1,111 @@
+{{if eq .PackageDescriptor.Package.Type "arch"}}
+	<h4 class="ui top attached header">{{.locale.Tr "packages.installation"}}</h4>
+	<div class="ui attached segment">
+		<div class="ui form">
+
+			<div class="field">
+				<label>{{svg "octicon-note"}} {{.locale.Tr "packages.arch.pacmanconf"}}</label>
+				<div class="markup"><pre class="code-block"><code>[{{.PackageDescriptor.Owner.LowerName}}.{{.PackageDescriptor.Metadata.BaseDomain}}]
+Server = https://{{.PackageDescriptor.Metadata.BaseDomain}}/api/packages/{{.PackageDescriptor.Owner.LowerName}}/arch/archlinux/x86_64</code></pre></div>
```

Comment:
This should probably use `{{AppSubUrl}}` Like in here https://github.com/go-gitea/gitea/blob/a12a5f3652c339b17b187ff424a480631a3c1e1e/templates/package/content/debian.tmpl#L7

Reply:
Yes, i was trying that before, but the result was empty URL. Do i have to add some additional parameters/apply specific condition for `{{AppSubUrl}}` variable to work properly?
![image](https://github.com/go-gitea/gitea/assets/58640116/1a1f6ed5-a114-46fc-9464-ffb6456887fd)

Reply:
Looks like it has to be part of [`gitea-origin-url`](https://github.com/go-gitea/gitea/blob/cdd3d4b8d8814137e16116002e5bf6aad38b794f/web_src/js/webcomponents/GiteaOriginUrl.js#L4) custom html element

```
diff --git a/templates/package/content/arch.tmpl b/templates/package/content/arch.tmpl
index 5f5c8cd49791..8d82265056bf 100644
--- a/templates/package/content/arch.tmpl
+++ b/templates/package/content/arch.tmpl
@@ -6,7 +6,7 @@
                        <div class="field">
                                <label>{{svg "octicon-note"}} {{.locale.Tr "packages.arch.pacmanconf"}}</label>
                                <div class="markup"><pre class="code-block"><code>[{{.PackageDescriptor.Owner.LowerName}}.{{.PackageDescriptor.Metadata.BaseDomain}}]
-Server = https://{{.PackageDescriptor.Metadata.BaseDomain}}/api/packages/{{.PackageDescriptor.Owner.LowerName}}/arch/archlinux/x86_64</code></pre></div>
+Server = <gitea-origin-url data-url="{{AppSubUrl}}/api/packages/{{.PackageDescriptor.Owner.LowerName}}/arch/archlinux/x86_64"></gitea-origin-url></code></pre></div>
                        </div>

                        <div class="field">
```

Reply:
The reason why there is a custom element is explained in the pull request https://github.com/go-gitea/gitea/pull/22861
TLDR: it's hard for server side to always know the correct user visible url so its calculated in the browser.

Reply:
@artizirk 
Corrected `arch.tmpl`, to use `{{AppURL}}` variable:
```html
<div class="markup"><pre class="code-block"><code>[{{.PackageDescriptor.Owner.LowerName}}.{{.PackageDescriptor.Metadata.BaseDomain}}]
SigLevel = Optional TrustAll
Server = {{AppUrl}}api/packages/{{.PackageDescriptor.Owner.LowerName}}/arch/archlinux/x86_64</code></pre></div>
</div>
```
I would also like to remove `{{.PackageDescriptor.Metadata.BaseDomain}}` variable from template. Is there any way to get domain name in UI?

Reply:
Changed template to use `{{.RegistryHost}}` variable, similarly with container registry.

```html
<div class="markup"><pre class="code-block"><code>[{{.PackageDescriptor.Owner.LowerName}}.{{.RegistryHost}}]
SigLevel = Optional TrustAll
Server = {{AppUrl}}api/packages/{{.PackageDescriptor.Owner.LowerName}}/arch/archlinux/x86_64</code></pre></div>
</div>
```